### PR TITLE
INFRA-821-add-event-bridge

### PR DESCRIPTION
**Pull request by jmicceri**

### CODE REVIEW

File: sink/event_bridge.go [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+	}
+
+	sess := session.Must(session.NewSession())
+	svc := eventbridge.New(sess)
+
+	req, resp := svc.DescribeEventBusRequest(&eventbridge.DescribeEventBusInput{
+		Name: aws.String(busName),
+	})
+
+	fmt.Println(resp)
```
Comment (lorenzo) [inline sink/event_bridge.go:40]: should be removed or converted to a `log.Debugf`

File: sink/event_bridge.go [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+	log.Infof("[sink/eventbridge] Starting writer")
+
+	for {
+		select {
+		case batch := <-s.batchCh:
+			entries := make([]*eventbridge.PutEventsRequestEntry, 0)
+
+			for _, data := range batch {
+				entry := &eventbridge.PutEventsRequestEntry{
+					EventBusName: aws.String("michael-test"),
```
Comment (lorenzo) [inline sink/event_bridge.go:135]: The bus name should be a field in the `EBSink` struct

File: sink/event_bridge.go [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+					Detail:       aws.String(string(data)),
+					DetailType:   aws.String("Nomad Firehose"),
+					Source:       aws.String("com.seatgeekadmin.NomadFirehose"),
```
Comment (lorenzo) [inline sink/event_bridge.go:138]: Should this also be configurable?

File: sink/event_bridge.go [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+				}
+
+				entries = append(entries, entry)
+				// id = id + 1
```
Comment (lorenzo) [inline sink/event_bridge.go:142]: remove line, but make sure that EB messages don't require a unique id per message

File: sink/event_bridge.go [hunk 5]
```diff
@@ -0,0 +1,178 @@
...
+			}
+
+			err := s.sendBatch(entries)
+			if err != nil && strings.Contains(err.Error(), "AWS.SimpleQueueService.BatchRequestTooLong") {
```
Comment (lorenzo) [inline sink/event_bridge.go:146]: You are matching on an `SQS` error, try to find out if there is anything similar for EB

File: sink/event_bridge.go [hunk 6]
```diff
@@ -0,0 +1,178 @@
...
+		}
+	}
+}
+
+func (s *EBSink) sendBatch(entries []*eventbridge.PutEventsRequestEntry) error {
+	req, _ := s.eventbridge.PutEventsRequest(&eventbridge.PutEventsInput{
+		Entries: entries,
+	})
+	err := req.Send()
+	if err == nil { // resp is now filled
```
Comment (lorenzo) [inline sink/event_bridge.go:173]: misleading comment

**Review by lorenzo — CHANGES_REQUESTED** [review]

File: sink/event_bridge.go (hunk 1 above)
Comment (jmicceri) [reply sink/event_bridge.go]: converted

File: sink/event_bridge.go (hunk 3 above)
Comment (jmicceri) [reply sink/event_bridge.go]: added these to env vars / the struct

File: sink/event_bridge.go (hunk 4 above)
Comment (jmicceri) [reply sink/event_bridge.go]: Removed, nope it doesn't require a unique ID

File: sink/event_bridge.go (hunk 5 above)
Comment (jmicceri) [reply sink/event_bridge.go]: I removed the additional check for the SQS error text. There isn't really batch functionality for EB so there wasn't a comparable error

File: sink/event_bridge.go (hunk 6 above)
Comment (jmicceri) [reply sink/event_bridge.go]: removed comment

File: sink/event_bridge.go [hunk 7]
```diff
@@ -0,0 +1,180 @@
...
+					Detail:       aws.String(string(data)),
+					DetailType:   aws.String(s.detailType),
+					Source:       aws.String(s.source),
+				}
+
+				entries = append(entries, entry)
+			}
+
+			err := s.sendBatch(entries)
+			if err != nil {
```
Comment (lorenzo) [inline sink/event_bridge.go:148]: If you don't know if there is any limit for the payload, or don't know how to detect the error, maybe remove this block of code?

Comment (jmicceri) [reply sink/event_bridge.go:148]: heh actually had it removed then added it back in, will remove

**Review by lorenzo — APPROVED** [review]

**Pull request merged**